reqwest-retry/src/middleware.rs

```diff
@@ -178,11 +178,24 @@ where
 
                         n_past_retries += 1;
                         continue;
-                    } else {
-                        result
                     }
                 }
-                Some(_) | None => result,
+                Some(_) | None => {}
+            };
+
+            // Report whether we failed with or without retries.
+            break if n_past_retries > 0 {
+                result.map_err(|err| {
+                    Error::Middleware(
```

Comment:
this looks wrong, on this code path, this is not a middleware error, it's a request error. 

a good example of a middleware error is when the middleware cannot do something it needs to do to function, like at like 146 above.

the `result` variable is the result of actually making a request and it failing or succeeding somehow, so these errors are `request` errors, not middleware errors.

Reply:
to make this work you need to change `ReqwestMiddleware::Error` so that the `Reqwest` variant contains your retry data

Reply:
I can't change `reqwest::Error` since it's part of the `reqwest` crate and not of `request-middleware` crate, so we need to use the dynamic error variant here.

Reply:
you can change `reqwest_middleware::Error` , in fact you already did in this PR. that's what i'm suggesting

Reply:
That would require changing `reqwest_middleware::Error` so that it has a variant specific to retry errors, this would break the isolation between the general retry stack and the specific middleware for retrying.

Reply:
yeah i see now, that's annoying